app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt

```diff
@@ -26,19 +67,140 @@ class SplashActivityTest {
     SplashActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
   )
 
-  @Before
-  fun setUp() {
-    Intents.init()
+  @Test
+  fun testSplashActivity_initialOpen_routesToOnBoardingActivity() {
+    activityTestRule.launchActivity(null)
+    intended(hasComponent(OnBoardingActivity::class.java.name))
   }
 
   @Test
-  fun testSplashActivity_initialOpen_routesToHomeActivity() {
-    activityTestRule.launchActivity(null)
-    intended(hasComponent(ProfileActivity::class.java.name))
+  fun testSplashActivity_secondOpen_routesToChooseProfileActivity() {
+    simulateAppAlreadyOnBoarded()
+    launch(SplashActivity::class.java).use {
+      intended(hasComponent(ProfileActivity::class.java.name))
+    }
   }
 
-  @After
-  fun tearDown() {
-    Intents.release()
+  private fun simulateNewAppInstance() {
+    // Simulate a fresh app install by clearing any potential on-disk caches using an isolated on-boarding flow controller.
+    createTestRootComponent().getOnBoardingFlowController().clearOnBoardingFlow()
+    onIdle()
+  }
+
+  private fun simulateAppAlreadyOnBoarded() {
+    // Simulate the app was already onboarded by creating an isolated on-boarding flow controller and saving the onboarding status
```

Comment:
In this same sentence `on-boarding` has  been used differently. Suggest making it consistent here and everywhere else.

Reply:
Done